### PR TITLE
[TF Led] Fix wrong decoder attention mask behavior

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR fixes TF LED. I wrongly added some lines to TFLed that automatically change the attention mask. However, this is incorrect behavior and not present in the PT version of the model. Sadly, I discovered this now after the release yesterday. @LysandreJik do you think we can patch this fix to circumvent breaking backward compatibility (but it's a bug IMO anyway).

This also fixes consequencetly the flaky `let_pt_tf_equivalence` test. I ran the test 40 times and it does not fail anymore.
## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/led/modeling_tf_led.py [hunk 1]
```diff
@@ -1862,28 +1862,16 @@ def call(
...
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask = None
         if input_shape[-1] > 1:
             combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
         else:
             combined_attention_mask = _expand_mask(
                 tf.ones((input_shape[0], input_shape[1] + past_key_values_length)), tgt_len=input_shape[-1]
             )
 
-        if inputs["attention_mask"] is None and inputs["input_ids"] is not None and input_shape[-1] > 1:
```
Comment (patrickvonplaten) [inline src/transformers/models/led/modeling_tf_led.py:1873]: I wrongly copied this from the old Bart templates. Those lines previously automatically create a correct attention mask in case there are pad_tokens. However, we do not support this behavior in the PyTorch version of LED and should not support it. In case some inputs should be padded the tokenizers should take care of masking the inputs.

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Review by LysandreJik — APPROVED** [review]
LGTM! Will do a patch release.
